src/main/java/io/projectriff/invoker/server/GrpcServerAdapter.java

```diff
@@ -138,6 +146,47 @@ private OutputSignal fromSpringMessage(Tuple2<Integer, Message<byte[]>> out) {
                 ;
     }
 
+    // Used to transform the publisher chain into one that doesn't forward cancel() calls once it has complete()d.
+    private Function<? super Publisher<Tuple2<Integer, Message<byte[]>>>, ? extends Publisher<Tuple2<Integer, Message<byte[]>>>> ignoreCancelsAfterComplete() {
+        return Operators.lift((f, actual) ->
+                new CoreSubscriber<>() {
```

Comment:
there is a default method `currentContext`, but it should be implemented here and delegate to `actual.currentContext()`, so that the context is preserved